src/vivarium_cluster_tools/utilities.py

```diff
@@ -7,6 +7,11 @@
 import sys
 
 from loguru import logger
+from typing import Dict, List, Tuple
+try:
+    from pip._internal.operations import freeze
```

Comment:
Document why you need this try/except

Reply:
added comment explaining where freeze is depends on version of pip